src/components/table/bl-table.ts

```diff
@@ -0,0 +1,285 @@
+import { CSSResultGroup, html, LitElement, PropertyValues, TemplateResult } from "lit";
+import { customElement, property, state } from "lit/decorators.js";
+import "element-internals-polyfill";
+import { event, EventDispatcher } from "../../utilities/event";
+import style from "./bl-table.css";
+import BlTableCell from "./table-cell/bl-table-cell";
+import BlTableHeaderCell from "./table-header-cell/bl-table-header-cell";
+import BlTableRow from "./table-row/bl-table-row";
+
+export const blTableTag = "bl-table";
+
+export const blSortChangeEventName = "bl-table-sort";
+export const blRowSelectChangeEventName = "bl-table-row-select";
+
+/**
+ * @tag bl-table
+ * @summary Baklava Table component
+ *
+ */
+@customElement(blTableTag)
+export default class BlTable extends LitElement {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  /**
+   * Selected table row selection key list
+   */
+  @property({ type: Array, reflect: true, attribute: "select-value" })
+  get selectValue(): string[] {
+    return this._selectValue;
+  }
+  set selectValue(value: string[]) {
+    this._selectValue = value;
+    this.updateComplete.then(() => {
+      this.querySelectorAll("bl-table-header-cell,bl-table-cell,bl-table-row").forEach(com => {
+        (com as BlTableHeaderCell | BlTableCell | BlTableRow).requestUpdate();
+      });
+    });
+  }
+
+  /**
+   * Sets table row as selectable
+   */
+  @property({ type: Boolean, reflect: true })
+  selectable = false;
+
+  /**
+   * Sets table row multiple selection enable
+   */
+  @property({ type: Boolean, reflect: true })
+  multiple = false;
+
+  /**
+   * Sets table as sortable
+   */
+  @property({ type: Boolean, reflect: true })
+  sortable = false;
+  /**
+   * Sets table first column as sticky
+   */
+  @property({ type: Boolean, reflect: true, attribute: "sticky-first-column" })
+  stickyFirstColumn = false;
+  /**
+   * Sets table last column as sticky
+   */
+  @property({ type: Boolean, reflect: true, attribute: "sticky-last-column" })
+  stickyLastColumn = false;
+
+  /**
+   * Sets table sorted column key
+   */
+  @property({ type: String, reflect: true, attribute: "sort-key" })
+  get sortKey(): string {
+    return this._sortKey;
+  }
+  set sortKey(value: string) {
+    this._sortKey = value;
+  }
+
+  /**
+   * Sets table sorting direction
+   */
+  @property({ type: String, reflect: true, attribute: "sort-direction" })
+  get sortDirection(): string {
+    return this._sortDirection;
+  }
+  set sortDirection(value: string) {
+    this._sortDirection = value;
+  }
+
+  /**
+   * Fires when table sort options changed
+   */
+  @event("bl-table-sort") private onSort: EventDispatcher<string[]>;
+
+  /**
+   * Fires when selected table rows changed
+   */
+  @event("bl-table-row-select") private onRowSelect: EventDispatcher<string[]>;
```

Comment:
Same as mentioned previously, we can remove the table prefix from the event name here.